### PR TITLE
Add read support from BigQuery

### LINKED ISSUE #7: Add remote read support
## What happens?

Currently, data can get sent to remote BigQuery table, but once there, the data cannot be read back out of the table via a Prometheus query.

## What were you expecting to happen?

Remote data should be available to Prometheus and any "downstream" analysis tools (Grafana, Alertmanager, etc.) to maximize utility.

## Additional comments:

Can reference influxdb example in the official Prometheus repo for an example of "remote read": https://github.com/prometheus/prometheus/blob/master/documentation/examples/remote_storage/remote_storage_adapter/influxdb/client.go#L115

### PR DESCRIPTION
**Pull request by Smiley73**
## Description

Add remote read support to this storage adapter.

Fixes #7 

## Type of change

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: bigquerydb/client.go [hunk 1]
```diff
@@ -180,3 +184,155 @@ func (c *BigqueryClient) Describe(ch chan<- *prometheus.Desc) {
...
+	}
+	labelPairs = append(labelPairs, &prompb.Label{
+		Name:  model.MetricNameLabel,
+		Value: row["metricname"].(string),
+	})
+	metric[model.LabelName(model.MetricNameLabel)] = model.LabelValue(row["metricname"].(string))
+	return &prompb.Sample{Timestamp: row["timestamp"].(time.Time).Unix(), Value: row["value"].(float64)}, metric, labelPairs, nil
+}
+
+// orderTimeSeries sole purpose is to make Prometheus happy
```
Comment (danschdatsci) [inline bigquerydb/client.go:317]: Could I get a little more detail? If we have "ORDER BY" in the SQL query, I'm curious what you ran into that was making prometheus angry.

Comment (Smiley73) [reply bigquerydb/client.go:317]: The other storage adapters all have a separate sorting algorithm and the `order by` is leftover code from my initial experimenting. I removed it and we're consistent with the others now.

File: bigquerydb/client.go [hunk 2]
```diff
@@ -180,3 +184,155 @@ func (c *BigqueryClient) Describe(ch chan<- *prometheus.Desc) {
...
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		iter, err := query.Read(ctx)
+		level.Debug(c.logger).Log("msg", "BiQuery query", "rows received", iter.TotalRows)
+		defer cancel()
+
+		if err != nil {
+			return nil, err
+		}
+
+		tsList, err := rowsToTimeseries(iter)
```
Comment (seanmalloy) [inline bigquerydb/client.go:207]: Missing the check for `err != nil` here.

File: bigquerydb/client.go [hunk 3]
```diff
@@ -180,3 +184,155 @@ func (c *BigqueryClient) Describe(ch chan<- *prometheus.Desc) {
...
+			Value: value.(string),
+		})
+		metric[model.LabelName(name)] = model.LabelValue(value.(string))
+	}
+	labelPairs = append(labelPairs, &prompb.Label{
+		Name:  model.MetricNameLabel,
+		Value: row["metricname"].(string),
+	})
+	metric[model.LabelName(model.MetricNameLabel)] = model.LabelValue(row["metricname"].(string))
+	return &prompb.Sample{Timestamp: row["timestamp"].(time.Time).Unix(), Value: row["value"].(float64)}, metric, labelPairs, nil
```
Comment (seanmalloy) [inline bigquerydb/client.go:314]: Since this is the only `return` statement in this function do you think it makes sense to return an always `nil` error?

Comment (seanmalloy) [reply bigquerydb/client.go:314]: The error could just be remvoved from the function signature.

**Pull request merged**